src/utils/data-hooks.ts

```diff
@@ -391,105 +325,137 @@ export const runQueriesWithHooks = async <T>(
     throw new Error(message);
   }
 
+  const queryList: {
+    definition: Query;
+    result: unknown;
+    diffForIndex?: number;
+  }[] = queries
+    .map((query, index) => {
+      const details = { definition: query, result: EMPTY };
+
+      // If data hooks are enabled, we want to send a separate `get` query for
+      // every `set` query (in the same transaction), so that we can provide the
+      // data hooks with a "before and after" of the modified records.
+      //
+      // The version of the record *after* the modification is already available
+      // without the extra `get` query, since `set` queries return the modified
+      // record afterward, but in order to get the version of the record
+      // *before* the modification, we need a separate `get` query.
```

Comment:
```suggestion
      // If data hooks are enabled, we want to send a separate `get` query for
      // every `set` query (in the same transaction), so that we can provide the
      // data hooks with a "before and after" of the modified records.
      // The version of the record *after* the modification is already available
      // without the extra `get` query, since `set` queries return the modified
      // record afterward, but in order to get the version of the record
      // *before* the modification, we need a separate `get` query.
```

I personally find empty lines in line-comments weird to look at. Personal preference, absolutely not blocking on that.

Reply:
Agreed that they might be "weird to look at", but not having them makes the comment harder to read. Just like, in a book, every new context is a new paragraph, the same is true for code comments. Otherwise, if the comment gets as long as the one above, it IMO becomes unreadable.